Redux store

notlfish opened the pull request:
### What?
Pull request to configure the redux store and connect the react components to the store.

### Before Merging
The following tasks are to be completed before merging:
- [x] Add `books` duck.
- [x] Configure the store with the corresponding book reducer.
- [x] Add add book functionality. Book `title` and `author` are taken from the form. `id` is autogenerated. `category`, `chapters`, and `currentChapter` are hardcoded for this pull request.
- [x] Render books from the store. Delete hardcoded books.
- [x] Add remove book functionality.

notlfish committed:
Configure store for books. Add add book functionality

notlfish committed:
Update lock files

notlfish committed:
Add remove book functionality

Davidosky007 commented on package.json:
```diff
@@ -13,6 +13,8 @@
+    "uuid": "^8.3.2",
     "web-vitals": "^1.0.1"
   },
   "scripts": {
```
* Kindly remove this `ESLINT_NO_DEV_ERRORS=true` from the start script as it makes the project difficult to start, if there  is a reason for the command kindly update your readme file with instructions on how to properly start the project 🙏

notlfish replied:
There is a reason for it to be there. It prevents linting errors from being treated as actual errors and crashing the app, which is _very_ annoying behavior.
I'm curious, though, what kind of problems does it give you? What system did you try to run the app on?

Anyway, I did change the start script back to the default.

Davidosky007 reviewed (changes requested):
## Hi @notlfish 👋, 

Good job so far! 💪 
* Descriptive commit messages ✔️ 
* Well written readme file ✔️ 
* No linter errors ✔️ 
* Professional PR ✔️ 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

notlfish committed:
Change package.json start script back to default.

Add dev-start script for running the server while treating linting
errors as warnings.

eliasecasta reviewed (approved):
## Status:  Approved 🟢

Hi, @notlfish, 

I tried to test both start cases without any of them failing for me or giving issues, maybe @Davidosky007 knows some future issues that this may cause and asked for the change, I tried to answer your question without luck, good luck in your next milestones!

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
